src/modules/rlm_kafka/rlm_kafka.c

```diff
@@ -377,29 +512,56 @@ static ssize_t kafka_xlat(void *instance, REQUEST *request, char const *fmt, cha
 	entry = rbtree_finddata(inst->rkt_by_name_tree, &my_topic);
 	if (!entry || !entry->rkt) {
 		RWARN("No configuration section exists for kafka topic \"%s\"", buf);
-		return -1;
+		goto error;
+	}
+
+	/*
+	 *  Extract the header specification, and generate the headers
+	 *
+	 */
+	q = p;
+	if (*p == '(') {
+		p = strchr(p, ')');
+		if (!p) {
+			REDEBUG("Header list is missing closing parenthesis)");
+			goto error;
+		}
+		MEM(headers = talloc_strndup(NULL, q + 1, p - q - 1));
+		if (*headers && create_headers(request, headers, &hdrs) < 0) {
+			RDEBUG3("Failed to create headers");
```

Comment:
`REDEBUG`? At least this feels like it should show up in `-X`.

Reply:
Agreed.